glib/src/subclass/types.rs

```diff
@@ -247,6 +247,8 @@ pub trait ObjectSubclass: Sized + 'static {
     /// Optional.
     const ABSTRACT: bool = false;
 
+    type Type: ObjectType + crate::object::ObjectSubclassIs<Subclass=Self>;
+
     /// Parent Rust type to inherit from.
     type ParentType: ObjectType
```

Comment:
We already know the `ParentType` because of how `glib_wrapper!` works. Can we make use of that for making sure the types align correctly (without adding a new trait)?

Reply:
Then that would work with any glib wrapper that has the same parent, even if it's a different type, right? I think we want to enforce that `ObjectSubclass::Type` is exactly a wrapper around the `ObjectSubclass`, not a wrapper around it's parent, a type with the same parent, or a child. An unsafe marker trait like this seems to be a good way to do this.

Also I think it would be possible to have blanket implementations for `IsA<Object>`, `IsA<Widget>` using this, based on whether or not the `ObjectSubclass` implements `ObjectImpl`, `WidgetImpl`, etc. So it wouldn't be necessary to specify the parent types when defining the wrapper. The problem with that is that we couldn't have implementations that conflict or could theoretically conflict. (So we basically couldn't have blanket implementations of `IsA` for anything else.)

Reply:
Not being able to have any other blanket impls seems ok to me. This seems like it would be the most important blanket impl ;) I'm not sure how that could work though, but if you find a way to make it work.

Apart from parent classes there are also implemented interfaces to consider here btw. `gio::ReadStream` for example implements `gio::Seekable`, if you're looking for an example.